### ISSUE TITLE
Query: Apply Include only once on given entity

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#16899]
```C#
[ConditionalTheory]
        [MemberData(nameof(IsAsyncData))]
        public virtual Task Include_on_GroupJoin_SelectMany_DefaultIfEmpty_with_coalesce_result4(bool isAsync)
        {
            var expectedIncludes = new List<IExpectedInclude>
            {
                new ExpectedInclude<Gear>(g => g.Weapons, "Weapons"),
                new ExpectedInclude<Officer>(g => g.Weapons, "Weapons")
            };

            return AssertIncludeQuery<Gear>(
                isAsync,
                gs => from g1 in gs.Include(g => g.Weapons)
                      join g2 in gs.Include(g => g.Weapons)
                          on g1.LeaderNickname equals g2.Nickname into grouping
                      from g2 in grouping.DefaultIfEmpty()
                      select new
                      {
                          g1,
                          g2,
                          coalesce = g2 ?? g1
                      },
                expectedIncludes,
                elementSorter: e => e.g1.Nickname,
                clientProjections: new List<Func<dynamic, object>>
                {
                    r => r.g1,
                    r => r.g2,
                    r => r.coalesce
                });
        }
```
It causes collection join to be applied twice. Which causes duplicated entry in collection. This happens only for non tracking query. Tracking query resolve to same entity instance.

**Comment by smitpatel** [comment]
Duplicate of #15783
